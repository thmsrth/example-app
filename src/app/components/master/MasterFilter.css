.MasterFilter {
  width: 100%;
  padding-bottom: var(--section-padding);
}

.MasterFilterInput {
  padding-right: calc(var(--section-padding) * 2);
  padding-bottom: var(--section-padding);
}
