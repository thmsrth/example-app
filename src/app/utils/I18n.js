const translations = {
  en: {
    answer: {
      no: 'No',
      yes: 'Yes',
    },
    actions: {
      back: 'Back',
      cancel: 'Cancel',
      create: 'Add',
      delete: 'Delete',
      detail: 'Detail',
      disabled: 'Disabled action',
      drink: 'Drink',
      edit: 'Edit',
      ok: 'OK',
      save: 'Save',
    },
    beers: {
      actions: {
        loading: 'Loading beers...',
      },
      states: {
        empty: 'No data available.',
        error: {
          400: 'Bad Request',
          404: 'Not found',
          422: 'Unprocessable Entities',
          500: 'Internal server error',
          loading: 'An error has occurred while fetching the beers.',
        },
      },
      title: 'Beers that go well with',
    },
    data: {
      actions: {
        adding: 'Creating a new food ...',
        deleting: 'Deleting food ...',
        loading: 'Loading data ...',
        updating: 'Updating food ...',
      },
      attribute: {
        checkbox: {
          title: 'Checkbox',
        },
        createdAt: {
          title: 'Created at',
        },
        createdBy: {
          title: 'Created by',
        },
        collection: {
          fields: {
            phone: {
              title: 'Phone',
              pattern: /^\+?\d*$/,
              validationAlert: 'Please enter a valid phone number',
            },
            date: {
              title: 'Date',
            },
          },
          title: 'Collection',
        },
        calories: {
          title: 'Calories',
        },
        fat: {
          title: 'Fat',
        },
        id: {
          title: 'ID',
        },
        language: {
          none: 'No language detected',
          title: 'Language',
        },
        name: {
          title: 'Name',
        },
        multiselect: {
          title: 'Multiselect',
        },
        nutritionalInformation: {
          title: 'Nutritional information',
        },
        protein: {
          title: 'Protein',
        },
        priceAmount: {
          pattern: /^[0-9]+(\.[0-9]{0,2})?$/,
          title: 'Amount',
          validationAlert: 'Please enter a valid amount (2 digits decimal)',
        },
        priceCurrencyCode: {
          title: 'Currency',
        },
        radio: {
          title: 'Radio',
        },
        status: {
          title: 'Status',
        },
        text: {
          title: 'Text',
        },
        updatedAt: {
          title: 'Updated at',
        },
        updatedBy: {
          title: 'Updated by',
        },
      },
      confirmDelete: 'Do you really want to delete this entry ?',
      confirmDrink: 'Do you really want to drink this beer ?',
      entity: 'Food',
      infos: {
        title: 'Infos',
      },
      pagination: {
        next: 'Next',
        prev: 'Prev',
        total: '%{count} results',
      },
      title: 'Food',
      states: {
        empty: 'No data available.',
        error: {
          400: 'Bad Request',
          404: 'Not found',
          422: 'Unprocessable Entities',
          500: 'Internal server error',
          adding: 'An error has occurred while adding a new food',
          deleting: 'An error has occurred while deleting the food',
          loading: 'An error has occurred while fetching the data',
          updating: 'An error has occurred while updating the food',
        },
      },
    },
    dateFormat: 'DD/MM/YYYY',
    filter: {
      all: 'All',
      dateFormat: 'yyyy-mm-dd',
      dateMin: '2012-01-01',
      dateMax: '2099-12-31',
      datePattern: '[0-9]{4}-[0-9]{2}-[0-9]{2}',
      from: 'From:',
      to: 'To:',
    },
    form: {
      validationAlert: 'The following entry must be valid',
      noOptionSelected: 'Nothing selected',
    },
    languages: {
      actions: {
        loading: 'Loading languages...',
      },
      states: {
        empty: 'No data available.',
        error: {
          400: 'Bad Request',
          404: 'Not found',
          422: 'Unprocessable Entities',
          500: 'Internal server error',
          loading: 'An error has occurred while fetching the languages.',
        },
      },
    },
    route: {
      noAccess: 'It seems that you do not have the necessary access for this page. Please check your permissions.',
      notFound: 'Unfortunately, the requested route does not exist.',
    },
  },
};

class I18n {
  static t(keyString, options = {}) {
    function findTranslation(keys, translationObject, keyPath) {
      if (!keys.length) {
        // Replace options values in translation
        return Object.keys(options).reduce((translation, key) => {
          const option = new RegExp(`%{${key}}`, 'g');
          return translation.replace(option, options[key]);
        }, translationObject);
      }

      if (Object.keys(translationObject).some(key => key === keys[0])) {
        return findTranslation(keys.slice(1), translationObject[keys[0]], `${keyPath}.${keys[0]}`);
      }
      throw new Error(`Missing translation "${keys[0]}" in key "${keyPath}"`);
    }

    return findTranslation(keyString.split('.'), translations, 'translations');
  }
}

export default I18n;


export function getErrorStatusTranslation(e, translationPath) {
  const status = e.request && e.request.status;
  const knownErrorsStatus = [400, 404, 422];
  if (status && knownErrorsStatus.some(errorStatus => errorStatus === status)) {
    return I18n.t(`${translationPath}.${status}`);
  }
  return I18n.t(`${translationPath}.500`);
}

export function getErrorMessage(e, translationPath) {
  if (e.response
    && e.response.data
    && Object.prototype.hasOwnProperty.call(e.response.data, 'errorMessage')) {
    return e.response.data.errorMessage;
  }
  return getErrorStatusTranslation(e, translationPath);
}
