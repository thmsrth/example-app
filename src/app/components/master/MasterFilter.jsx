import React from 'react';

import PropTypes from 'prop-types';
import Filter from '../../../global/components/Filter';
import dataItemType from '../types/dataItem';
import filterType from '../../../global/components/Filter/types/filter';
import filtersValuesType from '../../../global/components/Filter/types/filtersValues';

import I18n from '../../utils/I18n';

import './MasterFilter.css';

const MasterFilter = ({
  data, dataFilters, filters, handleFilterItems, fetchFilteredItems,
}) => (
  <Filter
    className="MasterFilter"
    items={data}
    filters={filters}
    dataFilters={dataFilters}
    handleFilterItems={handleFilterItems}
    fetchFilteredItems={fetchFilteredItems}
    serverSidedFiltering
  />
);

export default MasterFilter;

MasterFilter.propTypes = {
  data: PropTypes.arrayOf(dataItemType),
  dataFilters: filtersValuesType,
  filters: PropTypes.arrayOf(filterType),
  handleFilterItems: PropTypes.func,
  fetchFilteredItems: PropTypes.func,
};

MasterFilter.defaultProps = {
  data: [],
  dataFilters: {},
  filters: [
    {
      name: 'id',
      title: `${I18n.t('en.data.attribute.id.title')}:`,
      type: 'number',
    },
    {
      name: 'name',
      title: `${I18n.t('en.data.attribute.name.title')}:`,
      type: 'text',
    },
    {
      name: 'createdAt',
      title: 'Created At:',
      type: 'datetime',
      range: true,
      dateMin: I18n.t('en.filter.dateMin'),
      dateMax: I18n.t('en.filter.dateMax'),
      pattern: I18n.t('en.filter.datePattern'),
      placeholder: I18n.t('en.filter.dateFormat'),
    },
    {
      name: 'createdBy',
      title: 'Created by:',
      type: 'text',
    },
    {
      name: 'updatedAt',
      title: 'Updated At:',
      type: 'datetime',
      range: true,
      dateMin: I18n.t('en.filter.dateMin'),
      dateMax: I18n.t('en.filter.dateMax'),
      pattern: I18n.t('en.filter.datePattern'),
      placeholder: I18n.t('en.filter.dateFormat'),
    },
    {
      name: 'updatedBy',
      title: 'Updated by:',
      type: 'text',
    },
  ],
  handleFilterItems: () => {},
  fetchFilteredItems: () => {},
};
