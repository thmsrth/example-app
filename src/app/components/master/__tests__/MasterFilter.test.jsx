import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import staticData from '../../../static/data.json';
import MasterFilter from '../MasterFilter';
import I18n from '../../../utils/I18n';

global.permissions = {};

const { _embedded } = staticData;

const props = {
  data: _embedded.foods,
};

describe('MasterFilter', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<MasterFilter {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
