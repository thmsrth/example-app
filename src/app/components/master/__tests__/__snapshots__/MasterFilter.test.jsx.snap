// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MasterFilter should render the component correctly 1`] = `
<FilterWrapper
  className="MasterFilter"
  dataFilters={Object {}}
  fetchFilteredItems={[Function]}
  filters={
    Array [
      Object {
        "name": "id",
        "title": "ID:",
        "type": "number",
      },
      Object {
        "name": "name",
        "title": "Name:",
        "type": "text",
      },
      Object {
        "dateMax": "2099-12-31",
        "dateMin": "2012-01-01",
        "name": "createdAt",
        "pattern": "[0-9]{4}-[0-9]{2}-[0-9]{2}",
        "placeholder": "yyyy-mm-dd",
        "range": true,
        "title": "Created At:",
        "type": "datetime",
      },
      Object {
        "name": "createdBy",
        "title": "Created by:",
        "type": "text",
      },
      Object {
        "dateMax": "2099-12-31",
        "dateMin": "2012-01-01",
        "name": "updatedAt",
        "pattern": "[0-9]{4}-[0-9]{2}-[0-9]{2}",
        "placeholder": "yyyy-mm-dd",
        "range": true,
        "title": "Updated At:",
        "type": "datetime",
      },
      Object {
        "name": "updatedBy",
        "title": "Updated by:",
        "type": "text",
      },
    ]
  }
  handleFilterItems={[Function]}
  items={
    Array [
      Object {
        "calories": 266,
        "createdAt": "2018-10-28T14:48:54.01293",
        "createdBy": "thomas",
        "fat": 34,
        "id": 1,
        "name": "Pizza",
        "protein": 0.6,
        "status": "UNHEALTHY",
        "updatedAt": null,
        "updatedBy": null,
      },
      Object {
        "calories": 508,
        "createdAt": "2018-11-28T04:48:54.01293",
        "createdBy": "tommy",
        "fat": 1,
        "id": 2,
        "name": "Sushi",
        "protein": 26.7,
        "status": "HEALTHY",
        "updatedAt": null,
        "updatedBy": null,
      },
      Object {
        "calories": 271,
        "createdAt": "2018-12-28T14:48:54.01293",
        "createdBy": "ben",
        "fat": 1.3,
        "id": 3,
        "name": "Steak",
        "protein": 6.6,
        "status": "UNHEALTHY",
        "updatedAt": "2018-12-29T12:48:54.01293",
        "updatedBy": "thomas",
      },
    ]
  }
  serverSidedFiltering={true}
/>
`;
